common/include/serial_port.h

```diff
@@ -0,0 +1,65 @@
+/*
+ * serial_port.h
+ *
+ *  Created on: 2020��9��18��
+ *      Author: duanning
+ */
+
+#ifndef INCLUDE_SERIAL_PORT_H_
+#define INCLUDE_SERIAL_PORT_H_
+
+#include <iostream>
+#include <boost/asio.hpp>
+#include <boost/bind.hpp>
+#include "abstract_device.h"
+#include "abstract_protocol.h"
+
+using namespace boost::asio;
+
+class SerialPort: public AbstractDevice {
+public:
+	enum character_size_type {character_size_5 = 5, character_size_6 = 6, \
+		character_size_7 = 7, character_size_8 = 8};
+	enum parity_type {parity_none = 0, parity_odd = 1, parity_even = 2};
+	enum stop_bits_type {stop_bits_one, stop_bits_onepointfive, stop_bits_two};
+	enum flow_control_type {flow_control_none, flow_control_software, flow_control_hardware};
+
+    static const constexpr char* kSerialPortName 	= "serial_port_name";
+    static const constexpr char* kSerialPortData = "serial_port_data";
+
+    SerialPort(const string& name);
+    SerialPort(const string& name, AbstractProtocol& protocol);
+    SerialPort(string name, int baud_rate, character_size_type character_size, \
+    		parity_type parity, stop_bits_type stop_bits, flow_control_type flow_control);
+    virtual ~SerialPort();
+
+	virtual error_condition ReConfig(const CommonKeyPairs& config);
+	virtual error_condition DeConfig();
+	error_condition Set(const string& key, const unsigned int value);
+	error_condition Set(const string& key, \
+				const vector<uint8_t>& value);
+	error_condition Get(const string& key, unsigned int& result);
+	error_condition Get(const string& key, \
+				vector<uint8_t>& result);
+
+private:
+	io_service io_s_;
+	serial_port serial_port_;
+	AbstractProtocol application_protocol_;
+	boost::system::error_code ec;
+
+	struct {
```

Comment:
Please consider using "key pairs" to set parameter. E.g. we could call SerialPortDevice::Set("boardrate", 115200).